src/core/meshcop/commissioner.cpp

```diff
@@ -88,7 +88,7 @@ Commissioner::Commissioner(Instance &aInstance)
 
 void Commissioner::SetState(otCommissionerState aState)
 {
-    VerifyOrExit(mState != aState);
+    SuccessOrExit(Get<Notifier>().Update(mState, aState, OT_CHANGED_COMMISSIONER_STATE));
 
     otLogInfoMeshCoP("Commissioner State: %s -> %s", StateToString(mState), StateToString(aState));
```

Comment:
need a local variable to store the old state when the state changes for the log here?